integration-tests/bats/remotes.bats

```diff
@@ -2020,6 +2020,44 @@ SQL
     [[ "$output" =~ "Everything up-to-date." ]] || false
 }
 
+@test "remotes: call dolt_checkout track flag sets upstream" {
+    mkdir remote
+    mkdir repo1
+
+    cd repo1
+    dolt init
+    dolt remote add origin file://../remote
+    dolt sql -q "CREATE TABLE a (pk int)"
+    dolt commit -Am "add table a"
+    dolt push --set-upstream origin main
+    dolt checkout -b other
+    dolt push --set-upstream origin other
+
+    cd ..
+    dolt clone file://./remote repo2
+
+    cd repo2
+    dolt branch
+    [[ ! "$output" =~ "other" ]] || false
+
+    run dolt sql << SQL
+    call dolt_checkout('--track', 'origin/other');
```

Comment:
Would be nice to test another form of using `--track`, such as if a local branch name is supplied or to test the error case where no arg supplied to `--track` should return an error message.